### PR TITLE
frontend: wai-aria

### PR DESCRIPTION
**Pull request by nguyend-nam**
#### What's this PR do?

- [x] Add basic information about WAI-ARIA

### CODE REVIEW

File: Engineering/Frontend/WAI-ARIA.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+tags: engineering/frontend, wai-aria, accessibility
+author: Nguyen Dinh Nam
+date: 2022-09-30
+---
+
+Since modern websites and applications contain huge amounts of dynamic content and complex components with no semantics to describe what they mean, users with **no mouse** equiped or those dependent on technologies such as **[Screen Reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related?hl=en)** might not be able to interact with those components.
```
Comment (ngolapnguyen) [inline Engineering/Frontend/WAI-ARIA.md:7]: A bit weird intro :kek: Using a computer with no mouse is not usual, or touch devices don't need mouse. I think we can just state directly the role of WAI-ARIA: to assist users with disabilities.

For users that like to navigate the web without using the mouse, I think it's just a matter of preference.

File: Engineering/Frontend/WAI-ARIA.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Web Accessibility Initiative’s Accessible Rich Internet Applications or simply **WAI-ARIA** is a technology that can help with such problems.
+
+## What is WAI-ARIA?
+Before semantic elements like `<nav>` or `<footer>` were introduced that define specific features of a web page, some developers would rely on JavaScript libraries that generate a bunch of **nested `<div>`s**, then styled them with CSS and controlled with JavaScript. 
+
+Even though they work fine, screen readers cannot make any sense of what those components are, no semantics provided so that they can assist their users. WAI-ARIA has **roles**, **states** and **properties** to solve this problem, giving an opportunity to add attributes to content and components that make them **meaningful** and enhance **accessibility**.
```
Comment (ngolapnguyen) [inline Engineering/Frontend/WAI-ARIA.md:14]: You might want to add a small list here, listing out the next 3 things that you are going to introduce to the readers. E.g.:

```
3 main features of WAI-ARIA:
- Roles
- Properties
- States
```

You can add some short description (or ref link) to each item. This is a way to summarize & let the readers know in advance what they should expect to see next.

File: Engineering/Frontend/WAI-ARIA.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+### Roles:
+[WAI-ARIA roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) define **what that HTML is or does**, add the required additional information in cases the markup does not provide the required role. If we write `<a role="button">`, the screen reader will recognize it as a button.
```
Comment (ngolapnguyen) [inline Engineering/Frontend/WAI-ARIA.md:17]: You might want to expand more on what a screen reader is (e.g. more ref link). Many readers may not know what it is yet.

Comment (ngolapnguyen) [reply Engineering/Frontend/WAI-ARIA.md:17]: Keep in mind that we want to introduce **screen reader** as a concept, not just any extension.

**Review by ngolapnguyen — CHANGES_REQUESTED** [review]
Good article overall 👍 I've put down some improvement that I think may be helpful.

File: Engineering/Frontend/WAI-ARIA.md (hunk 1 above)
Comment (tienan92it) [reply Engineering/Frontend/WAI-ARIA.md:7]: This intro and a few below paraphrases mention **Screen Reader** but it's only one of wai-aria targets. It should aim to **Assistive Technologies** like screen readers, text-to-speech software, alternate input technologies...

**Review by tienan92it — APPROVED** [review]
LGTM

**Review by ngolapnguyen — APPROVED** [review]

**Review by zlatanpham — APPROVED** [review]
@monotykamary can you publish the note?

**Pull request merged**